### PR TITLE
CIRCLES-272: fixed and simplified getStructure code

### PR DESCRIPTION
**Pull request by Peedee2002**
fixed silly bug and simplied the getStructure code

### CODE REVIEW

**Comment by Snuzzn** [comment]
Looks good. One small thing - is it possible to put a space in between the words "FlexEducation" and "GeneralEducation" just like "Computing Electives" is? 
I could hardcode this in the frontend but I thought it would be easier to do in the backend.

**Review by Snuzzn — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,7 +33,6 @@ jobs:
         echo "MONGODB_DATABASE=mongodb" >> env/backend.env
         echo "MONGO_INITDB_ROOT_USERNAME=github-ci" >> env/mongodb.env
         echo "MONGO_INITDB_ROOT_PASSWORD=github-ci" >> env/mongodb.env
-        docker compose up backend -d --build
```
Comment (Peedee2002) [inline .github/workflows/ci.yml:36]: technically not needed because this is run by the next command

**Comment by Peedee2002** [comment]
added some more misc fixes, such as raw_requirements.

**Comment by Snuzzn** [comment]
I think one of your recent changes reverted the change to the general program requirements because it's empty again.

**Request**: http://localhost:8000/programs/getStructure/3778/COMPA1

image

**Comment by Peedee2002** [comment]
@Snuzzn yup, i accidentally supressed the wrong keyerror. Fixed again, so ill just merge now.

**Pull request merged**